mlflow/telemetry/events.py

```diff
@@ -86,6 +86,48 @@ def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
         return {"is_prompt": _is_prompt(tags)}
 
 
+class CreateDatasetEvent(Event):
+    name: str = "create_dataset"
+
+
+class MergeRecordsEvent(Event):
+    name: str = "merge_records"
+
+    @classmethod
+    def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
+        if not isinstance(arguments, dict):
+            return None
+
+        records = arguments.get("records")
+        if records is None:
+            return None
+
+        input_type = type(records).__name__.lower()
+        if "dataframe" in input_type:
+            input_type = "pandas"
+        elif isinstance(records, list) and records:
+            first_elem = records[0]
```

Comment:
```suggestion
        elif isinstance(records, list):
            if records:
                first_elem = records[0]
                ...
            else:
                input_type = "list"
```
adding another `elif isinstance(records, list)` below looks a bit less readable